environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -61,6 +62,8 @@ public class EnvironmentCreationService {
 
     private final LoadBalancerEntitlementService loadBalancerEntitlementService;
 
+    private final EnvironmentRecipeService recipeService;
```

Comment:
I wouldn't inject it here, I would validate it through `EnvironmentValidatorService`. Maybe even create a separate class for the validation